commander/txs_tracking.go

```diff
@@ -0,0 +1,44 @@
+package commander
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+)
+
+func (c *Commander) txsTracking(txsHashChan <-chan common.Hash) error {
+	for {
+		select {
+		case <-c.workersContext.Done():
+			return nil
+		case txHash := <-txsHashChan:
+			err := c.waitUntilTxMinedAndCheckForFail(txHash)
+			if err != nil {
+				panic(err)
+			}
+		default:
+			time.Sleep(time.Millisecond * 300)
+		}
+	}
+}
+
+func (c *Commander) waitUntilTxMinedAndCheckForFail(txHash common.Hash) error {
+	tx, _, err := c.client.Blockchain.GetBackend().TransactionByHash(context.Background(), txHash)
+	if err != nil {
+		return errors.WithStack(err)
+	}
```

Comment:
Can we perhaps push `types.Transaction` structs to the channel instead of transaction hashes to avoid this call to the blockchain?